src/lib/operators/get_table.cpp

```diff
@@ -1,27 +1,63 @@
 #include "get_table.hpp"
 
+#include <algorithm>
 #include <memory>
+#include <sstream>
 #include <string>
+#include <unordered_set>
 #include <vector>
 
 #include "storage/storage_manager.hpp"
+#include "types.hpp"
 
 namespace opossum {
 
-GetTable::GetTable(const std::string& name) : _name(name) {}
+GetTable::GetTable(const std::string& name) : _name(name), _excluded_chunks() {}
+
+GetTable::GetTable(const std::string& name, std::vector<ChunkID> excluded_chunks)
+    : _name(name), _excluded_chunks(excluded_chunks) {}
 
 const std::string GetTable::name() const { return "GetTable"; }
 
 const std::string GetTable::description(DescriptionMode description_mode) const {
   const auto separator = description_mode == DescriptionMode::MultiLine ? "\n" : " ";
-  return name() + separator + "(" + table_name() + ")";
+  std::stringstream stream;
+  stream << name() << separator << "(" << table_name() << ")";
+  if (!_excluded_chunks.empty()) {
+    stream << separator << "(" << _excluded_chunks.size() << " Chunks pruned)";
+  }
+  return stream.str();
 }
 
 const std::string& GetTable::table_name() const { return _name; }
 
+const std::vector<ChunkID>& GetTable::excluded_chunks() const { return _excluded_chunks; }
+
 std::shared_ptr<AbstractOperator> GetTable::recreate(const std::vector<AllParameterVariant>& args) const {
-  return std::make_shared<GetTable>(_name);
+  return std::make_shared<GetTable>(_name, _excluded_chunks);
 }
 
-std::shared_ptr<const Table> GetTable::_on_execute() { return StorageManager::get().get_table(_name); }
+std::shared_ptr<const Table> GetTable::_on_execute() {
+  auto original_table = StorageManager::get().get_table(_name);
+  if (_excluded_chunks.empty()) {
+    return original_table;
+  }
+
+  // we create a copy of the original table and don't include the excluded chunks
+  auto pruned_table = Table::create_with_layout_from(original_table, original_table->max_chunk_size());
+  auto excluded_chunks_set = std::unordered_set<ChunkID>(_excluded_chunks.cbegin(), _excluded_chunks.cend());
+  for (ChunkID chunk_id{0}; chunk_id < original_table->chunk_count(); ++chunk_id) {
+    if (excluded_chunks_set.count(chunk_id)) {
+      continue;
+    }
+    pruned_table->emplace_chunk(original_table->get_chunk(chunk_id));
+  }
+  return pruned_table;
+}
+
+void GetTable::set_excluded_chunks(const std::vector<ChunkID>& excluded_chunks) {
+  _excluded_chunks.clear();
```

Comment:
I thought set_excluded_chunks was supposed to go in favour of the constructor arg?

Reply:
Where was this discussed? This version is consistent with the TableScan.

Reply:
Ah, nevermind.

Reply:
Then the table/index_scan are unclean as well. mmpf

Reply:
I disagree. That is a special feature of those operators. Usually, when you think about constructing a scan, you wouldn't think about including/excluding chunks. I am not a fan of big, bloated constructors.

Reply:
https://www.youtube.com/watch?v=khUCcIri0rY